### ISSUE TITLE
Bun Support

### ISSUE THREAD

**Issue by aklinker1** [wxt-dev/wxt#222]
### Feature Request

Use bun for package management and running scripts.

- [x] Can bun run WXT?
   - ❌  `bun run --bun wxt` - See https://github.com/wxt-dev/wxt/issues/222#issuecomment-1806911119
   - 🟡  `bun run --bun wxt build` - logs warning (see https://github.com/wxt-dev/wxt/issues/222#issuecomment-1806911119) but seems to work
   - 🟡  `bun run --bun wxt zip` - logs warning (see https://github.com/wxt-dev/wxt/issues/222#issuecomment-1806911119) but seems to work
   - ✅  `bun run wxt` 
   - ✅ `bun run wxt build` 
   - ✅ `bun run wxt zip` 
- [x] ~~_Add bun to `wxt init`_~~ #233 
- [ ] #185 

#### Is your feature request related to a bug?

N/A

### What are the alternatives?

N/A

### Additional context

Decided to add this since [`nypm`](https://github.com/unjs/nypm/) includes support for detecting `bun`, and because it's a really fast script runner lol.

**Comment by dvlden** [comment]
This could be related?
[bun/issues/3137](https://github.com/oven-sh/bun/issues/3137)

I don't have source code of your CLI, but checking the `dist/cli.cjs` at all async/awaited calls it simply returns empty object.

**Comment by aklinker1** [comment]
Here's the `dist/cli.js` file, cursor is at where the stack trace says there's an uninitialized variable.

<img width="855" alt="Screenshot 2023-11-11 at 2 01 57 PM" src="https://github.com/wxt-dev/wxt/assets/10101283/5ad463b1-82af-4439-ba74-684c19fc266d">

**Comment by aklinker1** [comment]
Ok, so accessing `res.default` causes this error. Here I've added logs, and it prints out 3 and `res`, but not 4 and `res.default`. So this appears to be a problem with Jiti. I'll do a test against pure JITI and see if it's my fault or Jiti's fault.

<img width="1156" alt="Screenshot 2023-11-11 at 2 04 41 PM" src="https://github.com/wxt-dev/wxt/assets/10101283/2fd65618-0bc1-43e3-b877-088da37a8b3f">

**Comment by aklinker1** [comment]
OK, so `res.default` must be a getter, and it must be throwing an error while trying to import the entrypoint file. Jiti works just fine with Vite.

**Comment by aklinker1** [comment]
Jiti appears to have native support for Bun, and skips the required transform steps. After upgrading bun to 1.0.11, all the WXT commands fail when ran in Bun.

You can see the prepared code has imports on line 1, but the Bun code doesn't have those same imports. Basically, all the transforms I do to get the code running in Node are completely ignored with Bun.

<img width="1112" alt="Screenshot 2023-11-11 at 2 20 17 PM" src="https://github.com/wxt-dev/wxt/assets/10101283/fb1ab313-3459-4cff-ad8a-24547ff60a19">

**Comment by aklinker1** [comment]
Got that bug fixed, but ran into another. Now we're blocked by https://github.com/oven-sh/bun/issues/4130

```
$ bun run --bun wxt    

WXT 0.9.2                                                                                                            2:32:49 PM
✔ Started dev server @ http://localhost:3000                                                                        2:32:49 PM
ℹ Pre-rendering chrome-mv3 for development with Vite 4.5.0                                                          2:32:49 PM
[bun] Warning: worker_threads.Worker option "execArgv" is not implemented.

$ echo $?
0
```

**Comment by aklinker1** [comment]
Bun issue has been fixed, waiting for https://github.com/oven-sh/bun/issues/4130 to be released. Not gonna try out the main branch, just gonna wait.

**Comment by aklinker1** [comment]
Tested against the latest main branch, was super easy.

```sh
bun upgrade --canary
bunx wxt@latest init wxt-bun --pm bun --template vanilla
bun run --bun wxt build
bun run --bun wxt
```

Everything worked, no warnings! 🎉 🎉 

I'll close this issue once the the fix is released, I assume v1.0.22. v1.0.21 is the current release, and doesn't include https://github.com/oven-sh/bun/issues/4130.

**Comment by aklinker1** [comment]
Tested on v1.0.23, and it is working!

If you're using bun, you can update your `package.json` scripts to, putting `bunx --bun` in front of all the WXT commands:

```json
"scripts": {
    "dev": "bunx --bun wxt",
    "dev:firefox": "bunx --bun wxt -b firefox",
    "build": "bunx --bun wxt build",
    "build:firefox": "bunx --bun wxt build -b firefox",
    "zip": "bunx --bun wxt zip",
    "zip:firefox": "bunx --bun wxt zip -b firefox",
    "compile": "tsc --noEmit",
    "postinstall": "bunx --bun wxt prepare"
}
```
